### PR TITLE
Support context menus on transitive dependency tree items

### PR DESCRIPTION
**Pull request by drewnoakes**
Contributes to #5935.

Adds support for showing item-specific context menus and adds a default menu.

image

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/project-system/pull/6153)

### CODE REVIEW

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/AttachedCollections/AttachedCollectionItemBase.cs [hunk 1]
```diff
@@ -42,7 +42,8 @@ public abstract class AttachedCollectionItemBase
         private static readonly HashSet<Type> s_supportedPatterns = new HashSet<Type>
         {
             typeof(ITreeDisplayItem),
-            typeof(IBrowsablePattern)
+            typeof(IBrowsablePattern),
+            typeof(IContextMenuPattern)
```
Comment (davkean) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/AttachedCollections/AttachedCollectionItemBase.cs:46]: Should it return null for this pattern if ContextMenuController is null?

Comment (drewnoakes) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/AttachedCollections/AttachedCollectionItemBase.cs:46]: It works out the same in the end: http://index/?leftProject=Microsoft.VisualStudio.DesignTools.Diagnostics&leftSymbol=kx9opq4zj20s&file=Tool%5CUI%5CControls%5CTreeView%5CInteractionPatternProvider.cs&line=30

**Comment by davkean** [comment]
Do you know why the other commands don't show?

**Comment by drewnoakes** [comment]
> Do you know why the other commands don't show?

Pushed an update that brings in "Scope to This".

image

**Pull request merged**